core/src/main/resources/lib/form/select/select.js

```diff
@@ -30,13 +35,24 @@ function updateListBox(listBox,url,config) {
 
         if (originalOnSuccess!=undefined)
             originalOnSuccess(rsp);
-    },
-    config.onFailure = function(rsp) {
-        // deleting values can result in the data loss, so let's not do that
-//        var l = $(listBox);
-//        l.options[0] = null;
-    }
+    };
+    config.onFailure = function (rsp) {
+        l.removeClassName("select-ajax-pending");
+        status.innerHTML = rsp.responseText;
+        if (status.firstChild) {
+            status.firstChild.setAttribute('data-select-ajax-error', 'true')
+        }
+        Behaviour.applySubtree(status);
+        // deleting values can result in the data loss, so let's not do that unless instructed
+        var header = rsp.getResponseHeader('X-Jenkins-Select-Error');
+        if (header && "clear".toUpperCase() === header.toUpperCase()) {
```

Comment:
[toLocale___Case](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase) is the variants that would have i18n issues.

But looking at this again, I can remove one call by just changing the header value to lowercase